### ISSUE TITLE
Error when using with Telescope on a dev-only env

### ISSUE THREAD

**Issue by Braunson** [fruitcake/laravel-telescope-toolbar#39]
I've installed Telescope and set it up to only register and be used on the dev environment, no problems there.

The issue is when I install this Toolbar and **publish the config**, when running `composer install` on a non-dev env, it throws a Facade error. 

Digging into things, the cause is the config references Horizon classes `Laravel\Telescope\EntryType` in the `collectors` array in the toolbar config file that's published.

Any ideas on a way around this for a local only environment?

I am only loading Telescope and Toolbar packages on the local environment in `AppServiceProvider.php@register`

```php
        if ($this->app->isLocal()) {
            // Laravel Telescope
            $this->app->register(\Laravel\Telescope\TelescopeServiceProvider::class);
            $this->app->register(TelescopeServiceProvider::class);

            // Laravel Telescope Toolbar
            $this->app->register(\Fruitcake\TelescopeToolbar\ToolbarServiceProvider::class);
        }
```

**Comment by barryvdh** [comment]
Did you put in in your require-dev section?

**Comment by Braunson** [comment]
@barryvdh Yes, same with Laravel Horizon which it's config doesn't seem to complain about missing classes.

**Comment by AndrewFeeney** [comment]
It's a bit hacky, but I was able to workaround this issue, without deleting the published config by adding the following to the top of my `config/telescope-toolbar.php` file:

```php
<?php

use Laravel\Telescope\EntryType;

// Add this conditional before the existing return statement
if (! class_exists(EntryType::class)) {
    return [];
}

return [
  //  ...
];
```

**Comment by yoshiotobe** [comment]
I had a similar issue. When I deploy laravel for production with a deployer, it gets error below.

Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1

However, if I put a lines of code in config/telescope-toolbar.php that @AndrewFeeney wrote above, there is no errors.

Well, the file calls a class not existed in production. So, one has to delete config/telescope-toolbar.php or put the codes when deploying. It might be better if that codes are put in the first place.

**Comment by Oxicode** [comment]
Thx @AndrewFeeney

**Comment by Braunson** [comment]
@barryvdh Can we expect an official fix for this? I can put in a PR for one..

**Comment by fh32000** [comment]
any update @Braunson @barryvdh

**Comment by njoguamos** [comment]
I decided to remove the `telescope-toolbar` config file until the issue is resolved.

**Comment by m13v** [comment]
I hit the same wall, and the root cause is config load order. Laravel's LoadConfiguration bootstrapper reads every file in config/ before any service provider boots, so the isLocal() guard in AppServiceProvider never gets a chance to matter. The config is already parsed, and Telescope's classes aren't autoloaded on prod. This is the classic 'published config hard-references a package class' trap. What fixed it: in the published telescope-toolbar.php, either swap the EntryType:: constants for their plain string values ('request', 'query') or wrap the collectors array in a class_exists() guard.
